### ISSUE TITLE
Gating forms micro adjustments 

### ISSUE THREAD

**Issue by mzparacha** [hicommonwealth/commonwealth#5458]
Some micro adjustments were not done in https://github.com/hicommonwealth/commonwealth/pull/5045, this ticket is a followup to do that.

- [x] 1- In edit group mode, the "required requirements" value is undefined -- API also doesn't use this value 
- [x] 2- Some chain ids in the chain selector requirement have null ids -- Update them to be defined
- [x] 3- Topics are not editable -- Update them to be editable
- [x] 4- For Cosmos chain requirement the chain dropdown should only include cosmos base chains and vice versa for eth.
- [x] 5- Remove near + solana from the gating form
- [x] 6- Fix gating form style issue on Frack

**Comment by mzparacha** [comment]
for 1
>  1- In edit group mode, the "required requirements" value is undefined -- API also doesn't use this value

I mentioned [this](https://github.com/hicommonwealth/commonwealth/pull/5045#issuecomment-1772796415) in the gating form pr review 
<img width="965" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/51641047/946825bc-fd48-4f1b-aa37-f1b9af6c5f9e">

but somehow this question still got missed there.

@zakhap can you please confirm what would be the expected behaviour here?

**Comment by mzparacha** [comment]
2 is nearly done, i just need to find the `ethChainid` for polkadot. Confirming from team

>  2- Some chain ids in the chain selector requirement have null ids -- Update them to be defined

**Comment by mzparacha** [comment]
I am currently blocked on 1

>  1- In edit group mode, the "required requirements" value is undefined -- API also doesn't use this value

and working on 6 (6 is not blocked)

>  6- Fix gating form style issue on Frack

**Comment by zakhap** [comment]
> for 1
> 
> >

CC: @jnaviask @rbennettcw

For this section, the "required requirements" has two options (as seen in the designs):
1. ALL requirements must be satisfied for the community member to be in the Group
2. X of N many requirements must be satisfied for the community member to be in the Group (this functions "SOME" logic)

Is there anything else here that needs to be defined for the backend + front-end to align here?

**Comment by rbennettcw** [comment]
@mzparacha `required_requirements` is part of the `metadata` JSON object. It doesn't exist by default and must be explicitly set.

With the current code on master, you can implement `required_requirements` in the UI without blockage.

Additionally, here's a PR which applies the `required_requirements` value to the actual gating logic: https://github.com/hicommonwealth/commonwealth/pull/5460 (not a blocker for the UI though)

**Comment by mzparacha** [comment]
Thanks for the responses.

> With the current code on master, you can implement required_requirements in the UI without blockage.

This is already done, but atm the API doesn't wasn't returning it so I just wanted to know if it was something similar to what we did with the requirement condition (hardcoded -> 'more than' in frontend etc).


> Additionally, here's a PR which applies the required_requirements value to the actual gating logic: https://github.com/hicommonwealth/commonwealth/pull/5460 (not a blocker for the UI though)

didn't knew about this. Thanks.


So for me I will take #5460 diff locally to make the matching frontend changes for `required_requirements`.

cc: @rbennettcw @zakhap

**Comment by rbennettcw** [comment]
There are no frontend changes required for #5460– it's entirely just a backend update.

Regarding `required_requirements`, what isn't being returned? The `required_requirements
` value is returned when a group is created and when groups are queried:

<img width="1250" alt="CleanShot 2023-10-30 at 12 36 32@2x" src="https://github.com/hicommonwealth/commonwealth/assets/126922418/84a8eac8-6b9b-4845-a38e-5dc0291b20e8">

**Comment by rbennettcw** [comment]
Yeah by default, the `required_requirements` value is not returned unless it's set when the group is created. A default value of `0` can be used to represent "all requirements must be set"

**Comment by mzparacha** [comment]
hm, i didnt knew that, Thanks

**Comment by CowMuon** [comment]
@mzparacha #1 on list here can be marked done?

**Comment by mzparacha** [comment]
yes @CowMuon. I marked it done and pr is ready for review.
